src/net.md

```diff
@@ -876,6 +877,85 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-extract-mediawiki-links]: #ex-extract-mediawiki-links
+<a name="ex-extract-mediawiki-links"/>
+## Extract all unique links from a MediaWiki markup
+
+[![reqwest-badge]][reqwest] [![regex-badge]][regex] [![cat-net-badge]][cat-net]
+
+Pull the source of a Mediawiki page using [`reqwest::get`] and then
+find all entries of internal and external links in that source with
+[`Regex::captures_iter`].
+
+In Mediawiki syntax:
+* Internal links take a form of `[[Page]]` or `[[Page|link text]]`.
```

Comment:
I would try to minimize the info regarding the actual mediawiki syntax (ideally removing them all together).
I would just add external link to wikipedia site describing the link syntax.